solr/core/src/java/org/apache/solr/handler/loader/XMLLoader.java

```diff
@@ -333,12 +339,14 @@ public SolrInputDocument readDoc(XMLStreamReader parser) throws XMLStreamExcepti
           log.debug(message);
         }
       } else {
-        log.warn("XML element <doc> has invalid XML attr: {}", attrName);
+        if (!(NAME.equals(attrName) && forgiveNameAttr)) {
+          log.warn("XML element <doc> has invalid XML attr: {}", attrName);
+        }
       }
     }
 
     StringBuilder text = new StringBuilder();
-    String name = null;
+    String currentFieldName = null;
```

Comment:
Couldn't resist from make it more descriptive.

Reply:
Sure :)